### PR TITLE
[WIP] make absorber reach a specific damping value

### PR DESCRIPTION
**Pull request by PrometheusPi**
This pull request is a minimal change in the behavior of the absorber.

Currently `ABSORBER_STRENGTH` and `ABSORBER_CELLS` are independent and one might reach no damping if the first is too small compared to the second.

This pull request redefines the meaning of `ABSORBER_STRENGTH` to be the damping factor the be reached (exponentially). 

 - [ ] perform tests

### CODE REVIEW

File: src/picongpu/include/fields/FieldManipulator.hpp [hunk 1]
```diff
@@ -61,7 +61,8 @@ class FieldManipulator
                 uint32_t pos_or_neg = i % 2;
 
                 uint32_t thickness = ABSORBER_CELLS[direction][pos_or_neg];
-                float_X absorber_strength = ABSORBER_STRENGTH[direction][pos_or_neg];
+		PMacc::algorthm::math::Log log;
+                float_X absorber_strength = -1.0 * log( ABSORBER_STRENGTH[direction][pos_or_neg] ) thickness;
```
Comment (psychocoderHPC) [inline src/picongpu/include/fields/FieldManipulator.hpp]: A `*` before thickness ist missing

File: src/picongpu/include/fields/FieldManipulator.hpp [hunk 2]
```diff
@@ -61,7 +61,8 @@ class FieldManipulator
                 uint32_t pos_or_neg = i % 2;
 
                 uint32_t thickness = ABSORBER_CELLS[direction][pos_or_neg];
-                float_X absorber_strength = ABSORBER_STRENGTH[direction][pos_or_neg];
+
+                float_X absorber_strength = -1.0 * math::log( ABSORBER_MAXFACTOR[direction][pos_or_neg] ) * thickness;
```
Comment (psychocoderHPC) [inline src/picongpu/include/fields/FieldManipulator.hpp]: This should be `-1.0 * math::log( ABSORBER_MAXFACTOR[direction][pos_or_neg] ) / thickness;` ??

Comment (PrometheusPi) [inline src/picongpu/include/fields/FieldManipulator.hpp]: @psychocoderHPC You are off course complete right with that - I made an error in deriving it in my lab book :sad:

**Comment by PrometheusPi** [comment]
*offline discussion with* @psychocoderHPC:
the old and this implementation have the strongest damping first - we should check if an inital weaker damping is better.

**Comment by PrometheusPi** [comment]
The exponential damping towards a total damping in the end seems to be counter productive.
It appears to be best to slowly reduce the field thus that it is still reflected on the boundary of the simulation box. In the current version, too low damping factors cause a close to total reflection at the outer boundary. Too high values (which would be favored  by this pull request) result in a reflection at the damping zone. The best damping factor in `dev` lies around `1e-2`. Which corresponds to a maximum damping of `0.7614` in this pull request. Since this appears to be non intuitive, I would favour the current `dev` implementation. 

index
 electromagnetic field energy evolution with different damping factors **p** as defined in `dev`.
simulated laser, each step (in E) is a "reflection/damping" at either y=0 or y=1024 boundary.

**Comment by PrometheusPi** [comment]
The best damping is counter-intuitive with this new pull request - I will close this proposed code change.

**Comment by PrometheusPi** [comment]
as discussed with @psychocoderHPC a slow initial damping which grows stronger towards the outer walls might be an improvement.

**Pull request closed without merging**